diff -Naur netopeer2-1.1.76.org/CMakeLists.txt netopeer2-1.1.76/CMakeLists.txt
--- netopeer2-1.1.76.org/CMakeLists.txt	2021-06-02 18:18:52.000000000 +1000
+++ netopeer2-1.1.76/CMakeLists.txt	2021-06-16 10:01:50.630305874 +1000
@@ -227,6 +227,7 @@
         set(ENV{NP2_MODULE_PERMS} \"${MODULES_PERMS}\")
         set(ENV{NP2_MODULE_OWNER} \"${MODULES_OWNER}\")
         set(ENV{NP2_MODULE_GROUP} \"${MODULES_GROUP}\")
+        set(ENV{SYSREPOCTL_EXECUTABLE} \"${SYSREPOCTL}\")
         execute_process(COMMAND \"${SCRIPT_DIR}/setup.sh\" RESULT_VARIABLE SETUP_RES)
         if(NOT SETUP_RES EQUAL \"0\")
             message(FATAL_ERROR \" scripts/setup.sh failed: \${SETUP_RES}\")
@@ -238,6 +239,8 @@
 if(GENERATE_HOSTKEY)
     install(CODE "
         message(STATUS \"Generating a new RSA host key \\\"genkey\\\" if not already added...\")
+        set(ENV{OPENSSL_EXECUTABLE} \"${OPENSSL}\")
+        set(ENV{SYSREPOCFG_EXECUTABLE} \"${SYSREPOCFG}\")
         execute_process(COMMAND ${SCRIPT_DIR}/merge_hostkey.sh RESULT_VARIABLE MERGE_HOSTKEY_RES)
         if(NOT MERGE_HOSTKEY_RES EQUAL \"0\")
             message(FATAL_ERROR \" scripts/merge_hostkey.sh failed: \${MERGE_HOSTKEY_RES}\")
@@ -247,6 +250,7 @@
 if(MERGE_LISTEN_CONFIG)
     install(CODE "
         message(STATUS \"Merging default server listen configuration if there is none...\")
+        set(ENV{SYSREPOCFG_EXECUTABLE} \"${SYSREPOCFG}\")
         execute_process(COMMAND ${SCRIPT_DIR}/merge_config.sh RESULT_VARIABLE MERGE_CONFIG_RES)
         if(NOT MERGE_CONFIG_RES EQUAL \"0\")
             message(FATAL_ERROR \" scripts/merge_config.sh failed: \${MERGE_CONFIG_RES}\")
diff -Naur netopeer2-1.1.76.org/scripts/merge_config.sh netopeer2-1.1.76/scripts/merge_config.sh
--- netopeer2-1.1.76.org/scripts/merge_config.sh	2021-06-02 18:18:52.000000000 +1000
+++ netopeer2-1.1.76/scripts/merge_config.sh	2021-06-17 18:16:54.425699757 +1000
@@ -43,8 +43,7 @@
                     <client-authentication>
                         <supported-authentication-methods>
                             <publickey/>
-                            <passsword/>
-                            <other>interactive</other>
+                            <password/>
                         </supported-authentication-methods>
                         <users/>
                     </client-authentication>
diff -Naur netopeer2-1.1.76.org/scripts/merge_hostkey.sh netopeer2-1.1.76/scripts/merge_hostkey.sh
--- netopeer2-1.1.76.org/scripts/merge_hostkey.sh	2021-06-02 18:18:52.000000000 +1000
+++ netopeer2-1.1.76/scripts/merge_hostkey.sh	2021-06-17 18:00:14.932694732 +1000
@@ -39,13 +39,14 @@
 STDIN`
 
 # generate edit config
-CONFIG="<keystore xmlns=\"urn:ietf:params:xml:ns:yang:ietf-keystore\">
+CONFIG="<keystore xmlns=\"urn:ietf:params:xml:ns:yang:ietf-keystore\" xmlns:ct=\"urn:ietf:params:xml:ns:yang:ietf-crypto-types\">
     <asymmetric-keys>
         <asymmetric-key>
             <name>genkey</name>
-            <algorithm>rsa2048</algorithm>
+            <public-key-format>ct:ssh-public-key-format</public-key-format>
+            <private-key-format>ct:rsa-private-key-format</private-key-format>
             <public-key>$PUBKEY</public-key>
-            <private-key>$PRIVKEY</private-key>
+            <cleartext-private-key>$PRIVKEY</cleartext-private-key>
         </asymmetric-key>
     </asymmetric-keys>
 </keystore>"
diff -Naur netopeer2-1.1.76.org/src/netconf_server.c netopeer2-1.1.76/src/netconf_server.c
--- netopeer2-1.1.76.org/src/netconf_server.c	2021-06-02 18:18:52.000000000 +1000
+++ netopeer2-1.1.76/src/netconf_server.c	2021-06-17 19:12:59.160838488 +1000
@@ -28,29 +28,29 @@
 int
 np2srv_sr_get_privkey(const struct lyd_node *asym_key, char **privkey_data, NC_SSH_KEY_TYPE *privkey_type)
 {
-    struct lyd_node_leaf_list *alg = NULL, *privkey = NULL;
+    struct lyd_node_leaf_list *privkey_format = NULL, *privkey = NULL;
     struct lyd_node *node;
 
     /* find the nodes */
     LY_TREE_FOR(asym_key->child, node) {
-        if (!strcmp(node->schema->name, "algorithm")) {
-            alg = (struct lyd_node_leaf_list *)node;
-        } else if (!strcmp(node->schema->name, "private-key")) {
+        if (!strcmp(node->schema->name, "private-key-format")) {
+            privkey_format = (struct lyd_node_leaf_list *)node;
+        } else if (!strcmp(node->schema->name, "cleartext-private-key")) {
             privkey = (struct lyd_node_leaf_list *)node;
         }
     }
-    if (!alg || !privkey) {
+    if (!privkey_format || !privkey) {
         ERR("Failed to find asymmetric key information.");
         return -1;
     }
 
-    /* set algorithm */
-    if (!strncmp(alg->value.ident->name, "rsa", 3)) {
+    /* set private key format */
+    if (!strcmp(privkey_format->value.ident->name, "rsa-private-key-format")) {
         *privkey_type = NC_SSH_KEY_RSA;
-    } else if (!strncmp(alg->value.ident->name, "secp", 4)) {
+    } else if (!strcmp(privkey_format->value.ident->name, "ec-private-key-format")) {
         *privkey_type = NC_SSH_KEY_ECDSA;
     } else {
-        ERR("Unknown private key algorithm \"%s\".", alg->value_str);
+        ERR("Unknown private key format \"%s\".", privkey_format->value_str);
         return -1;
     }
 
diff -Naur netopeer2-1.1.76.org/src/netconf_server_ssh.c netopeer2-1.1.76/src/netconf_server_ssh.c
--- netopeer2-1.1.76.org/src/netconf_server_ssh.c	2021-06-02 18:18:52.000000000 +1000
+++ netopeer2-1.1.76/src/netconf_server_ssh.c	2021-06-17 22:50:51.144560299 +1000
@@ -279,7 +279,7 @@
         } else if (op == SR_OP_DELETED) {
             auth &= ~NC_SSH_AUTH_PUBLICKEY;
         }
-    } else if (!strcmp(node->schema->name, "passsword")) {
+    } else if (!strcmp(node->schema->name, "password")) {
         if (op == SR_OP_CREATED) {
             auth |= NC_SSH_AUTH_PASSWORD;
         } else if (op == SR_OP_DELETED) {
