diff -Naur netopeer2-1.1.34.org/CMakeLists.txt netopeer2-1.1.34/CMakeLists.txt
--- netopeer2-1.1.34.org/CMakeLists.txt	2020-06-03 00:20:56.000000000 +1000
+++ netopeer2-1.1.34/CMakeLists.txt	2020-06-08 18:51:45.001279898 +1000
@@ -207,6 +207,8 @@
         set(ENV{NP2_MODULE_PERMS} ${MODULES_PERMS})
         set(ENV{NP2_MODULE_OWNER} ${MODULES_OWNER})
         set(ENV{NP2_MODULE_GROUP} ${MODULES_GROUP})
+        set(ENV{SYSREPOCTL} ${SYSREPOCTL})
+        set(ENV{SYSREPOCFG} ${SYSREPOCFG})
         execute_process(COMMAND ${SCRIPT_DIR}/setup.sh)
     ")
 else()
@@ -215,12 +217,15 @@
 if(GENERATE_HOSTKEY)
     install(CODE "
         message(STATUS \"Generating a new RSA host key \\\"genkey\\\" if not already added...\")
+        set(ENV{OPENSSL} ${OPENSSL})
+        set(ENV{SYSREPOCTL} ${SYSREPOCTL})
         execute_process(COMMAND ${SCRIPT_DIR}/merge_hostkey.sh)
     ")
 endif()
 if(MERGE_LISTEN_CONFIG)
     install(CODE "
         message(STATUS \"Merging default server listen configuration if there is none...\")
+        set(ENV{SYSREPOCFG} ${SYSREPOCFG})
         execute_process(COMMAND ${SCRIPT_DIR}/merge_config.sh)
     ")
 endif()
diff -Naur netopeer2-1.1.34.org/scripts/merge_config.sh netopeer2-1.1.34/scripts/merge_config.sh
--- netopeer2-1.1.34.org/scripts/merge_config.sh	2020-06-03 00:20:56.000000000 +1000
+++ netopeer2-1.1.34/scripts/merge_config.sh	2020-06-08 18:49:25.830525066 +1000
@@ -3,7 +3,7 @@
 set -e
 
 # avoid problems with sudo path
-SYSREPOCFG=`su -c "which sysrepocfg" $USER`
+#SYSREPOCFG=`su -c "which sysrepocfg" $USER`
 KS_KEY_NAME=genkey
 
 # check that there is no listen/Call Home configuration yet
diff -Naur netopeer2-1.1.34.org/scripts/merge_hostkey.sh netopeer2-1.1.34/scripts/merge_hostkey.sh
--- netopeer2-1.1.34.org/scripts/merge_hostkey.sh	2020-06-03 00:20:56.000000000 +1000
+++ netopeer2-1.1.34/scripts/merge_hostkey.sh	2020-06-08 18:51:25.875431893 +1000
@@ -3,8 +3,8 @@
 set -e
 
 # avoid problems with sudo path
-SYSREPOCFG=`su -c "which sysrepocfg" $USER`
-OPENSSL=`su -c "which openssl" $USER`
+#SYSREPOCFG=`su -c "which sysrepocfg" $USER`
+#OPENSSL=`su -c "which openssl" $USER`
 
 # check that there is no SSH key with this name yet
 KEYSTORE_KEY=`$SYSREPOCFG -X -x "/ietf-keystore:keystore/asymmetric-keys/asymmetric-key[name='genkey']/name"`
diff -Naur netopeer2-1.1.34.org/scripts/setup.sh netopeer2-1.1.34/scripts/setup.sh
--- netopeer2-1.1.34.org/scripts/setup.sh	2020-06-03 00:20:56.000000000 +1000
+++ netopeer2-1.1.34/scripts/setup.sh	2020-06-08 18:49:25.834524454 +1000
@@ -7,7 +7,7 @@
 fi
 
 # avoid problems with sudo path
-SYSREPOCTL=`su -c "which sysrepoctl" $USER`
+#SYSREPOCTL=`su -c "which sysrepoctl" $USER`
 MODDIR=${NP2_MODULE_DIR}
 PERMS=${NP2_MODULE_PERMS}
 OWNER=${NP2_MODULE_OWNER}
